core/db/migrate/20150113002122_create_spree_promotion_codes.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSpreePromotionCodes < ActiveRecord::Migration
+  def change
+    create_table :spree_promotion_codes do |t|
+      t.references :promotion, index: true
+      t.string :value
+      t.datetime :starts_at
```

Comment:
for our business, cloning the promotion would be far more optimal for that use case than adding a code with different expiry dates. almost all of our dev-effort requests are expiry-date-for-code related, and the main value of this feature to us would be to make those requests more self-service but allowing the rules to live on the parent promotion and the codes just to be more than one way to apply that code